support-frontend/assets/components/amazonPayForm/amazonPayForm.tsx

```diff
@@ -0,0 +1,192 @@
+			design: {
+				designMode: 'responsive',
+			},
+			onPaymentSelect: () => {
+				props.setAmazonPayPaymentSelected(true);
+			},
+			onError: (error) => {
+				// The widget UI will display an error to the user, so we can just log it
+				logException(`Amazon Pay wallet error: ${error.getErrorMessage()}`);
+			},
+		};
+
+		if (props.contributionType !== 'ONE_OFF') {
+			new amazonPaymentsObject.Widgets.Wallet({
+				...baseWalletConfig,
+				onReady: (billingAgreement) => {
+					props.setAmazonPayBillingAgreementId(
+						billingAgreement.getAmazonBillingAgreementId(),
+					);
+				},
+				agreementType: 'BillingAgreement',
+			}).bind('WalletWidgetDiv');
+		} else {
+			new amazonPaymentsObject.Widgets.Wallet({
+				...baseWalletConfig,
+				amazonOrderReferenceId: props.amazonPay.orderReferenceId,
+				onOrderReferenceCreate: (orderReference) => {
+					props.setAmazonPayOrderReferenceId(
+						orderReference.getAmazonOrderReferenceId(),
+					);
+				},
+				agreementType: 'orderReference',
+			}).bind('WalletWidgetDiv');
+		}
+
+		props.setAmazonPayWalletIsStale(false);
+	};
+
+	const createConsentWidget = (
+		amazonPaymentsObject: AmazonPaymentsObject,
+		amazonBillingAgreementId: string,
+	): void => {
+		const consentConfig: ConsentConfig = {
+			amazonBillingAgreementId,
+			sellerId: getSellerId(props.isTestUser),
+			design: {
+				designMode: 'responsive',
+			},
+			onReady: (billingAgreementConsentStatus) => {
+				const amazonBillingAgreementConsentStatus =
+					billingAgreementConsentStatus.getConsentStatus() === 'true';
+				props.setAmazonPayBillingAgreementConsentStatus(
+					amazonBillingAgreementConsentStatus,
+				);
+			},
+			onConsent: (billingAgreementConsentStatus) => {
+				const amazonBillingAgreementConsentStatus =
+					billingAgreementConsentStatus.getConsentStatus() === 'true';
+				props.setAmazonPayBillingAgreementConsentStatus(
+					amazonBillingAgreementConsentStatus,
+				);
+			},
+			onError: (error) => {
+				logException(`Amazon Pay consent error: ${error.getErrorMessage()}`);
+			},
+		};
+
+		new amazonPaymentsObject.Widgets.Consent(consentConfig).bind(
+			'ConsentWidgetDiv',
+		);
+	};
+
+	useEffect(() => {
+		if (props.amazonPaymentsObject) {
+			createWalletWidget(props.amazonPaymentsObject);
+		}
+	}, [props.amazonPaymentsObject, props.amazonPay.walletIsStale]);
+
+	useEffect(() => {
+		if (
+			props.amazonPaymentsObject &&
+			props.amazonPay.amazonBillingAgreementId &&
+			props.contributionType !== 'ONE_OFF'
+		) {
+			createConsentWidget(
+				props.amazonPaymentsObject,
+				props.amazonPay.amazonBillingAgreementId,
+			);
+		}
+	}, [
+		props.amazonPaymentsObject,
+		props.amazonPay.amazonBillingAgreementId,
+		props.contributionType,
+	]);
+
+	if (props.amazonLoginObject && props.amazonPaymentsObject) {
+		trackComponentLoad('amazon-pay-wallet-loaded');
+		return (
+			<div>
+				{props.checkoutFormHasBeenSubmitted &&
+					!props.amazonPay.paymentSelected && (
```

Comment:
This will require a second pass before we actually turn the payment method on- `checkoutFormHasBeenSubmitted` is no longer used on the new checkout so we won't be able to check for this error state. I'd be happy to make some time to pair next week on getting this integrated with the new validation system.